awswrangler/distributed/ray/modin/s3/_read_parquet.py

```diff
@@ -25,17 +27,19 @@ def _read_parquet_distributed(  # pylint: disable=unused-argument
     s3_additional_kwargs: Optional[Dict[str, Any]],
     arrow_kwargs: Dict[str, Any],
 ) -> Union[pd.DataFrame, Iterator[pd.DataFrame]]:
-    dataset_kwargs = {}
-    if coerce_int96_timestamp_unit:
-        dataset_kwargs["coerce_int96_timestamp_unit"] = coerce_int96_timestamp_unit
-    dataset = read_datasource(
-        datasource=ArrowParquetDatasource(),
-        parallelism=parallelism,
-        use_threads=use_threads,
-        paths=paths,
-        schema=schema,
-        columns=columns,
-        dataset_kwargs=dataset_kwargs,
-        path_root=path_root,
-    )
+    try:
+        dataset = read_datasource(
+            datasource=ArrowParquetDatasource(),
+            parallelism=parallelism,
+            meta_provider=FastFileMetadataProvider(),
+            paths=paths,
+            path_root=path_root,
+            use_threads=use_threads,
+            table_schema=schema,
+            columns=columns,
+            coerce_int96_timestamp_unit=coerce_int96_timestamp_unit,
+        )
+        dataset.fully_executed()
```

Comment:
Is this `fully_executed` really needed?

Reply:
And why the try/catch is needed now?

Reply:
Yes, purely because of error handling. All blocks will be read at the next step anyway, but we'd like to catch and raise the wrapped errors